theano/gof/tests/test_vm.py

```diff
@@ -197,14 +197,19 @@ def time_linker(name, linker):
 def test_partial_function():
     import numpy as np
     from theano.tests import unittest_tools as utt
-    x = tensor.scalar('input')
-    y = x ** 2
-    f = theano.function([x], [y + 7, y - 9, y / 14.], mode=Mode(
-        optimizer=None, linker=vm.VM_Linker(allow_partial_eval=True)))
 
-    assert f(3, output_subset=[0, 1, 2]) == f(3)
-    assert f(4, output_subset=[0, 2]) == [f(4)[0], f(4)[2]]
-    utt.assert_allclose(f(5), np.array([32., 16., 1.7857142857142858]))
+    def check_partial_function(linker_name):
+        x = tensor.scalar('input')
+        y = x ** 2
+        f = theano.function([x], [y + 7, y - 9, y / 14.], mode=Mode(
+            optimizer=None, linker=linker_name))
```

Comment:
I implemented the handling of updates in partially evaluated functions. Apparently, that case was not working in either Stack nor CVM. In the code, I assumed that the outputs corresponding to the updates are after the regular outputs, but I couldn't find the piece of code that populates the outputs (it is somewhere in the `accept` function of linkers, but it seems quite convoluted). Am I right about this assumption?